posts/2024-10-17-Rust-1.82.0.md

```diff
@@ -0,0 +1,320 @@
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.82.0 stable
+
+### `cargo info`
+
+Cargo now has an [`info` subcommand](https://doc.rust-lang.org/nightly/cargo/commands/cargo-info.html) to display information about a package in the registry, fulfilling a [long standing request](https://github.com/rust-lang/cargo/issues/948) just shy of its tenth anniversary! Several third-party extensions like this have been written over the years, and this implementation was developed as [cargo-information](https://crates.io/crates/cargo-information) before merging into Cargo itself.
+
+For example, here's what you could see for `cargo info cc`:
+
+<pre>
+<b><span style="color:#0A0">cc</span></b> <b><span style="color:#0AA">#build-dependencies</span></b>
+A build-time dependency for Cargo build scripts to assist in invoking the native
+C compiler to compile native C code into a static archive to be linked into Rust
+code.
+<b><span style="color:#0A0">version:</span></b> 1.1.23 <b><span style="color:#A50">(latest 1.1.30)</span></b>
+<b><span style="color:#0A0">license:</span></b> MIT OR Apache-2.0
+<b><span style="color:#0A0">rust-version:</span></b> 1.63
+<b><span style="color:#0A0">documentation:</span></b> https://docs.rs/cc
+<b><span style="color:#0A0">homepage:</span></b> https://github.com/rust-lang/cc-rs
+<b><span style="color:#0A0">repository:</span></b> https://github.com/rust-lang/cc-rs
+<b><span style="color:#0A0">crates.io:</span></b> https://crates.io/crates/cc/1.1.23
+<b><span style="color:#0A0">features:</span></b>
+  jobserver = []
+  parallel  = [dep:libc, dep:jobserver]
+<b><span style="color:#0AA">note</span></b><b>:</b> to see how you depend on cc, run `<b><span style="color:#0AA">cargo tree --invert --package cc@1.1.23</span></b>`
+</pre>
+
+By default `cargo info` describes the package version in the local `Cargo.lock`, if any, but as you can see it will indicate when there's a newer version too, and `cargo info cc@1.1.30` would report on that. The feature list also indicates a `+` next to any that are enabled.
+
+### Apple target promotions
+
+#### macOS on ARM64 is now Tier 1
+
+The Rust target `aarch64-apple-darwin` for macOS on ARM64 (M1-family or later Apple Silicon CPUs) is now a tier 1 target, indicating our highest guarantee of working properly. As the [platform support](https://doc.rust-lang.org/stable/rustc/platform-support.html) page describes, every change in the Rust repository must pass full tests on every tier 1 target before it can be merged. This ARM64 target was added to tier 2 back in Rust 1.49, making it available in `rustup`, and this new milestone raises it on par with ARM64 Linux and the X86 macOS, Linux, and Windows targets.
```

Comment:
```suggestion
The Rust target `aarch64-apple-darwin` for macOS on AArch64 (also known as ARM64 or Apple Silicon) is now a tier 1 target, indicating our highest guarantee of working properly. As the [platform support](https://doc.rust-lang.org/stable/rustc/platform-support.html) page describes, every change in the Rust repository must pass full tests on every tier 1 target before it can be merged. This target was introduced as tier 2 back in Rust 1.49, making it available in `rustup`. This new milestone puts the `aarch64-apple-darwin` target on par with the AArch64 Linux and the X86 macOS, Linux, and Windows targets.
```